### DISCUSSION TITLE
Proposal to adopt new Versioning Scheme

### DISCUSSION THREAD

**Discussion by MotoLegacy** [pspdev/pspdev#60] [Ideas]
Since its initial commit to this GitHub repository in Oct 2010, the reported SDK version has been `1.0+beta2`. This is despite significant changes having been made the last decade to justify, at the very least, a versioning bump.

As opposed to a version bump however, changing the scheme may be better suited. [Semantic Versioning](https://semver.org/) comes immediately to mind as it has become a standard in software as of recent. SemVer allows for easily reporting the SDK build date, which would be ideal for better specifying SDK versions while still maintaining the original 1.0 beta2 status (if absolutely necessary).

An immediate use case for this would be to have our Doxygen documentation better be identified as the actively maintained one (as opposed to http://psp.jim.sh/pspsdk-doc/) by seeing a date printed on the header and at the landing page instead of identifying it from the small font footer.

Would love to have a discussion about this here, looking forward to a response.

**Comment by sharkwouter** [comment]
I think it's a good idea to have some versioning which changes with new builds. I feel it should be automatic, though, not manual. I would propose to use the current date or the commit number.

**Comment by MotoLegacy** [comment]
> I think it's a good idea to have some versioning which changes with new builds. I feel it should be automatic, though, not manual. I would propose to use the current date or the commit number.

Using the commit number solely takes away the ease of understanding I suggested being present in situations such as determining which documentation page is up to date. It also isn't standardized on its own, which is probably important we make an effort into doing and why I suggested SemVer. We can be SemVer compliant while using both the build date and version number, however. This can easily be automated via Actions.

**Comment by artart78** [comment]
I'm not sure semantic versioning makes much sense as the API is just Sony's and there won't be breaking changes or much evolution (except additions). I think the build date (or 1.0-date) would be nice.

**Comment by MotoLegacy** [comment]
> I'm not sure semantic versioning makes much sense as the API is just Sony's and there won't be breaking changes or much evolution (except additions). I think the build date (or 1.0-date) would be nice.

This isn't true though, breaking changes have occurred, notably when the SDK became more POSIX compliant — `valloc` used to be used to allocate video RAM, but since this name exists in the POSIX world, it was renamed to `vramalloc`. That is a fairly breaking change. 

There are also a lot of features the SDK sorta shoves together like `PSP_LARGE_MEMORY` that I can see becoming depreciated.

**Comment by spiritfader** [comment]
Is this something that has been considered? It's been almost a year since this was brought up initially. What's the status so far on this being implemented? <br/>
This is pretty much a necessity to facilitate easier troubleshooting for things that have broken by recent commits and just from updating to the latest SDK. This would also allow an easier documentation of concepts/samples according to what's changed in wiki and other formats without them blindly becoming depreciated. <br/>
Semantic versioning along with more verbose changelog (proper changelog in general) of alterations would work well for this, and rather than see it as something that would "be nice to have" I see it as a necessity for not only consistency but usability of the codebase going forward.

**Comment by diamant3** [comment]
Hello guys! I will transfer this to the pspdev repo and then convert it into a discussion. Feel free to continue.

**Comment by sharkwouter** [comment]
Okay, so since this was reported we have implemented the following:
- Regular automated releases using date based version numbers.
- Automatic changelog generation for releases.
- The [online SDK documentation](https://pspdev.github.io/pspsdk/) now lists the build date instead of version 1.0+beta2.
- Each copy of PSPDEV now includes a build.txt file with info on the specific commit that was build for each part of the toolchain, like commit message, commit date and commit sha.

I would say this makes the situation a whole lot better. I hope that addresses this topic. More ideas are welcome, though.

**Comment by sharkwouter** [comment]
@MotoLegacy is  there anything you are still missing in our versioning?
